reference/metadata_tables.rst

```diff
@@ -167,3 +173,65 @@ Citus manages shard health on a per-placement basis and automatically marks a pl
+|      Name      |         Type         |       Description                                                         |
++================+======================+===========================================================================+
+| nodeid         |         int          | | Auto-generated identifier for an individual node.                       |
++----------------+----------------------+---------------------------------------------------------------------------+
+| groupid        |         int          | | Identifier used to denote a group of 1 primary and 0 or more            |
+|                |                      | | secondaries, when the streaming replication model is used. By default   |
+|                |                      | | it is the same as the nodeid.                                           | 
++----------------+----------------------+---------------------------------------------------------------------------+
+|  nodename      |         text         | | DNS host name of the worker node PostgreSQL server hosting this shard   |
+|                |                      | | placement.                                                              |
++----------------+----------------------+---------------------------------------------------------------------------+
+| nodeport       |         int          | | Port number on which the worker node PostgreSQL server hosting this     |
+|                |                      | | shard placement is listening.                                           |
++----------------+----------------------+---------------------------------------------------------------------------+
+| noderack       |        text          | | Rack placement information for the worker node (if available).          |
++----------------+----------------------+---------------------------------------------------------------------------+
+
+::
+
+    SELECT * from pg_dist_node;
+     nodeid | groupid | nodename  | nodeport | noderack | hasmetadata 
+    --------+---------+-----------+----------+----------+-------------
+          1 |       1 | localhost |    12345 | default  | f
+          2 |       2 | localhost |    12346 | default  | f
+          3 |       3 | localhost |    12347 | default  | f
+    (3 rows)
+
+
+Colocation information table
+---------------------------------------
+
+The pg_dist_colocation table contains information about which tables are physically co-located together. When creating tables, the user can specify a table to be co-located physically with another one. Citus then ensures physical co-location of the shards, and can perform optimizations like co-located joins without data shuffling and support features like distributed rollups and enforcing foreign-key constraints. Co-location can also be inferred if the shard-counts, replication factor and partition-column types are the same for two different tables.
```

Comment:
Second feedback here is that it seems weird to lead with "users can specify a custom colocation group". I think the ordering should go something like:

  * Define what colocation is
  * Enumerate its benefits within Citus
  * Explain automatic colocation (matching count, factor, type)
  * Mention custom override option

Here's my best stab at addressing the above:

> The pg_dist_colocation table contains information about which tables' shards should be placed together, or _colocated_. When two tables are in the same colocation group, Citus ensures shards with the same partition values will be placed on the same worker nodes. This enables join optimizations, certain distributed rollups, and foreign key support. Shard colocation is inferred when the shard counts, replication factors, and partition column types all match between two tables; however, a custom colocation group may be specified when creating a distributed table, if so desired.